anvil/src/config.rs

```diff
@@ -818,131 +818,141 @@ impl NodeConfig {
                     )
                 };
 
-            let block = provider
-                .get_block(BlockNumber::Number(fork_block_number.into()))
-                .await
-                .expect("Failed to get fork block");
+                let block = provider
+                    .get_block(BlockNumber::Number(fork_block_number.into()))
+                    .await
+                    .expect("Failed to get fork block");
 
-            let block = if let Some(block) = block {
-                block
-            } else {
-                if let Ok(latest_block) = provider.get_block_number().await {
-                    panic!(
-                            "Failed to get block for block number: {fork_block_number}\nlatest block number: {latest_block}"
+                let block = if let Some(block) = block {
+                    block
+                } else {
+                    if let Ok(latest_block) = provider.get_block_number().await {
+                        let mut message = format!(
+                            "Failed to get block for block number: {fork_block_number}\n\
+                        latest block number: {latest_block}"
                         );
-                }
-                panic!("Failed to get block for block number: {fork_block_number}")
-            };
+                        // If the user is forking from an older block in a non-archive node, the
+                        // `get_block` call will return null.
+                        if fork_block_number < latest_block.as_u64() {
+                            message.push('\n');
+                            message.push_str(NON_ARCHIVE_NODE_WARNING);
```

Comment:
hmm maybe we can do this to keep it a bit more brief

```suggestion
                            message.push_str(format!("\n{}", NON_ARCHIVE_NODE_WARNING));
```